client_properties_validator.js

```diff
@@ -0,0 +1,69 @@
+const Ajv = require('ajv');
+const ajv = new Ajv();
+
+const defaultSchema = {
+  '$schema': 'http://json-schema.org/draft-07/schema',
+  'type': 'object',
+  'properties': {
+    'scalar.dl.client.cert_holder_id': {
+      'type': 'string',
+    },
+    'scalar.dl.client.cert_version': {
+      'type': 'number',
+    },
+    'scalar.dl.client.cert_pem': {
+      'type': 'string',
+    },
+    'scalar.dl.client.private_key_pem': {
+      'type': 'string',
+    },
+    'scalar.dl.client.server.host': {
+      'type': 'string',
+    },
+    'scalar.dl.client.server.port': {
+      'type': 'number',
+    },
+    'scalar.dl.client.server.privileged_port': {
+      'type': 'number',
+    },
+    'scalar.dl.client.tls.ca_root_cert_pem': {
+      'type': 'string',
+    },
+  },
+};
+
+/**
+ * A validator used to validate the client properties object
+ */
+class ClientPropertiesValidator {
```

Comment:
How about making it `ClientProperties` class and having methods like the following ?

```
class ClientProperites {
  constructor(properties, required) {
    // validate here
  }

  // getters 
  getCertHolderId() {
  }
  ...
} 
```

Also, maybe we can define constants (like CERT_HOLDER_ID) in this file.

Reply:
Oh, it's a good idea and a better OO.
I will do this.